okio/src/commonMain/kotlin/okio/internal/-Path.kt

```diff
@@ -159,6 +155,84 @@ internal inline fun Path.commonResolve(child: Path): Path {
   return buffer.toPath()
 }
 
+@ExperimentalFileSystem
+@Suppress("NOTHING_TO_INLINE")
+internal inline fun Path.commonRelativeTo(other: Path): Path {
+  require(isAbsolute && other.isAbsolute || isRelative && other.isRelative) {
+    "Paths of different roots cannot be relative to each other: $this and $other"
+  }
+
+  val thisSlash = slash
+  val otherSlash = other.slash
+  require(thisSlash == otherSlash) {
+    "Paths of different platforms cannot be relative to each other: $this and $other"
+  }
+
+  val slash = thisSlash ?: otherSlash ?: Path.DIRECTORY_SEPARATOR.toSlash()
+
+  // We look at the path both have in common.
+  var lastCommonByteIndex = 0
+  while (lastCommonByteIndex < bytes.size &&
+    lastCommonByteIndex < other.bytes.size &&
+    (bytes[lastCommonByteIndex] and 0xff) == (other.bytes[lastCommonByteIndex] and 0xff)
```

Comment:
I think the `and 0xff` bit isn’t strictly necessary here, because left and right sides will both be signed bytes.